src/charm.py

```diff
@@ -131,10 +174,14 @@ def _set_alerts(self, container):
         """
         container.remove_path(RULES_DIR, recursive=True)
 
-        for topology_identifier, rules_file in self.metrics_consumer.alerts().items():
-            filename = "juju_" + topology_identifier + ".rules"
+        self._push_alert_rules_group(container, self.metrics_consumer.alerts())
+        self._push_alert_rules_group(container, self.remote_write_provider.alerts())
+
+    def _push_alert_rules_group(self, container, alerts):
+        for group_name, group in alerts.items():
+            filename = "juju_" + group_name + ".rules"
```

Comment:
I think the spec is one file per topology identifier, not one file per group name
https://github.com/canonical/prometheus-k8s-operator/blob/c285372fa51c6ef031239d39c4c66c5255ca6a48/src/charm.py#L134-L136

Reply:
fixed